src/Tools/ExternalAccess/LegacyCodeAnalysis/LegacyCodeAnalysisVisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Diagnostics;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LegacyCodeAnalysis
+{
+    [Export(typeof(ILegacyCodeAnalysisVisualStudioDiagnosticAnalyzerService))]
```

Comment:
Are we generally going to do the interface plus concrete type approach for all of this, or just want to directly export the concrete type? Usually for internally-sort of things where we're less interested in hiding implementation we've just done concrete types.

No harm of course doing it this way other than just more files and more code.

Reply:
➡️ Submitted #35727 to document the patterns I've started to settle on